xdsl/dialects/tensor.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from typing_extensions import Self
+
+from xdsl.dialects.builtin import IndexType, TensorType
+from xdsl.ir import Attribute, Dialect, SSAValue
+from xdsl.irdl import IRDLOperation, irdl_op_definition, result_def, var_operand_def
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+
+
+@irdl_op_definition
+class EmptyOp(IRDLOperation):
+    name = "tensor.empty"
+
+    dynamic_sizes = var_operand_def(IndexType)
+
+    tensor = result_def(TensorType[Attribute])
+
+    def __init__(self, dynamic_sizes: Sequence[SSAValue], tensor_type: Attribute):
+        super().__init__(
+            operands=(dynamic_sizes,),
+            result_types=(tensor_type,),
+        )
+
+    def print(self, printer: Printer):
+        if self.dynamic_sizes:
+            printer.print_string("(")
+            printer.print_list(self.dynamic_sizes, printer.print_ssa_value)
```

Comment:
in my understanding, the format is `tensor.empty(*list of ssa values* : *list of types of the ssa values*) : () -> *tensor type*`
This means that there should be two print lists here, one to print the values, then another to print the types of the values. Have a look at how the return or yield ops are printed, they do the same thing

Reply:
My understanding was incorrect, thank you for investigating